### PR TITLE
Add dagster cloud

### PR DESCRIPTION
**Pull request by geoHeil**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by geoHeil** [comment]
confirm

**Comment by geoHeil** [comment]
how can we add pex? https://pypi.org/project/pex/ which is a prerequisite - it looks like it only marginally makes sense when we already bundle via conda is it possible to monkeypatch/remove this dependency on the fly in this build?

**Comment by geoHeil** [comment]
@conda-forge-admin can you give guidance on how to handle pex?

**Comment by geoHeil** [comment]
pex seems to be handled now.

```
/home/conda/staged-recipes/build_artifacts/dagster-cloud_1718886636409/work/conda_build.sh: line 4: 3,: No such file or directory

```

still needs to be fixed

**Comment by geoHeil** [comment]
> pex seems to be handled now.
> 
> ```
> /home/conda/staged-recipes/build_artifacts/dagster-cloud_1718886636409/work/conda_build.sh: line 4: 3,: No such file or directory
> ```
> 
> still needs to be fixed

how can we get more debug logs? it is a bit unclear to me what conda_build.sh is being executed - as you see in the history I tried to use my own but that was clashing

**Comment by tmirko** [comment]
confirm

File: recipes/dagster-cloud/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e  # Exit immediately if a command exits with a non-zero status.
+
+# Install pex using pip
+$PYTHON -m pip install pex<3,>=2.1.132
```
Comment (jaimergp) [inline recipes/dagster-cloud/build.sh:5]:
```suggestion
$PYTHON -m pip install "pex<3,>=2.1.132"
```

This is the error, but you can't install pip stuff here as part of the recipe. If pex is not on conda-forge, create a recipe for it and put it under `recipes/`. The infra will figure out the build order.

Comment (geoHeil) [reply recipes/dagster-cloud/build.sh]: Thanks! I applied your suggestion.

**Comment by geoHeil** [comment]
Windows build is failing due to pex depending on fcntl which is not available for windows.

- how can this be resolved?
- can we remove/monkey-patch to not require pex? (as conda is already packaging

**Comment by milicevica23** [comment]
add me as a maintainer

**Comment by geoHeil** [comment]
I have disabled windows to workaround the issue (for now)

**Comment by geoHeil** [comment]
@conda-forge/staged-recipes any chance to monkey patch the build around pex (remove it) to also support windows? As conda serves as the packager - the code around pex should not be required in case it is used from conda

File: recipes/dagster-cloud-cli/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/dagster-cloud-cli-{{ version }}.tar.gz
+  sha256: 0ff9f243f83d76a110a96027489790e486fe8eaf5b7fca39322fad8fd2ba71b5
+
+build:
+  entry_points:
+    - dagster-cloud = dagster_cloud_cli.entrypoint:app
+  #noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  skip: true  # [win]
```
Comment (ocefpaf) [inline recipes/dagster-cloud-cli/meta.yaml:18]: It is OK for noarch package to fails on Windows with a missing dependency. If that is the case here, we prefer noarch b/c it is faster to build and the package will be ready when the missing dependency is available on Windows.

Comment (geoHeil) [reply recipes/dagster-cloud-cli/meta.yaml]: ok - I just rolled back CI should be failing again on windows soon

File: recipes/dagster-cloud-cli/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/dagster-cloud-cli-{{ version }}.tar.gz
+  sha256: 0ff9f243f83d76a110a96027489790e486fe8eaf5b7fca39322fad8fd2ba71b5
+
+build:
+  entry_points:
+    - dagster-cloud = dagster_cloud_cli.entrypoint:app
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  #skip: true  # [win]
```
Comment (ocefpaf) [inline recipes/dagster-cloud-cli/meta.yaml:18]:
```suggestion
```

Comment (geoHeil) [reply recipes/dagster-cloud-cli/meta.yaml]: applied cleanup in all files

File: recipes/dagster-cloud-cli/meta.yaml (hunk 2 above)
Comment (geoHeil) [reply recipes/dagster-cloud-cli/meta.yaml]: changes are implemented - how can we move forward?

**Comment by ocefpaf** [comment]
The Windows failure is not a missing module it is:

```
ModuleNotFoundError: No module named 'fcntl'
```

That means we have to make these OS-noarch. Please see https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies

You will need to add the `- __unix  # [unix]` package in run and a `conda-forge.yml` file with:

```
noarch_platforms:
  - linux_64
  - osx_64
```

**Comment by geoHeil** [comment]
> The Windows failure is not a missing module it is:
> 
> ```
> ModuleNotFoundError: No module named 'fcntl'
> ```
> 
> That means we have to make these OS-noarch. Please see https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies
> 
> You will need to add the `- __unix # [unix]` package in run and a `conda-forge.yml` file with:
> 
> ```
> noarch_platforms:
>   - linux_64
>   - osx_64
> ```

ok but for now this module will also not come to conda (fast) as far as I understand.

and pex where this module is loaded transitively is also not required for running/using the package if it is packaged from conda. So would there be a way to get rid of that particular dependency somehow?

**Comment by geoHeil** [comment]
> The Windows failure is not a missing module it is:
> 
> ```
> ModuleNotFoundError: No module named 'fcntl'
> ```
> 
> That means we have to make these OS-noarch. Please see https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies
> 
> You will need to add the `- __unix # [unix]` package in run and a `conda-forge.yml` file with:
> 
> ```
> noarch_platforms:
>   - linux_64
>   - osx_64
> ```

@ocefpaf ` noarch packages can't have selectors. If the selectors are necessary, please remove noarch: python.` is what I get when I try to implement your suggestion

**Comment by ocefpaf** [comment]
> @ocefpaf ` noarch packages can't have selectors. If the selectors are necessary, please remove noarch: python.` is what I get when I try to implement your suggestion


Did you add the conda-forge.yml file as described in the docs?

**Comment by geoHeil** [comment]
> > @ocefpaf ` noarch packages can't have selectors. If the selectors are necessary, please remove noarch: python.` is what I get when I try to implement your suggestion
> 
> Did you add the conda-forge.yml file as described in the docs?

yes - see https://github.com/conda-forge/staged-recipes/blob/62e869635d883c006a32d8dc3e0af7edb392d393/recipes/pex/conda-forge.yaml

**Comment by ocefpaf** [comment]
> > > @ocefpaf ` noarch packages can't have selectors. If the selectors are necessary, please remove noarch: python.` is what I get when I try to implement your suggestion
> > 
> > 
> > Did you add the conda-forge.yml file as described in the docs?
> 
> yes - see https://github.com/conda-forge/staged-recipes/blob/62e869635d883c006a32d8dc3e0af7edb392d393/recipes/pex/conda-forge.yaml

I fixed the name for you in the latest commit. It should work now.

**Pull request merged**